src/charm.py

```diff
@@ -384,6 +423,14 @@ def _on_install(self, _event: InstallEvent) -> None:
 
         self.unit.status = ActiveStatus()
 
+    def _block_on_openstack_config(self) -> None:
```

Comment:
Perhaps this can also do the openstack cloud config check and return a boolean for the caller to determine whether they should exit the hook?

Reply:
Updated! thank you!